### PR TITLE
Leak fix

### PR DESCRIPTION
**Pull request by beeender**
SetLongArrayRegion will do a memcpy and queries might be leaked if
exception happens.

### CODE REVIEW

**Comment by beeender** [comment]
@realm/java

**Comment by kneth** [comment]
:+1:

File: changelog.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * Upgraded to NDK R10e. Using gcc 4.9 for all architectures.
  * Updated Realm Core to 0.95.6
    - Fixed a bug where an async query can be copied incomplete in rare cases (#1717).
+ * Fixed some memory leaks when using async query.
```
Comment (cmelchior) [inline changelog.txt]: `Fixed some` -> `Fixed potential memory leak`

**Comment by cmelchior** [comment]
Apart from small comment :+1:

**Pull request merged**